### ISSUE TITLE
php aritsan app:name should be removed or has to be enhanced drastically.

### ISSUE THREAD

**Issue by juslintek** [laravel/framework#18583]
- Laravel Version: 5.4.16
- PHP Version: 7.1
- Database Driver & Version: mysql mariadb 10.2.4

### Description:

I'm angry at the php artisan app:name

Would be a good command, but if it doesn't change App namespace to custom everywhere even in vendor, then it is of no use and you are stuck with strict App namespace, unless you want to build everything from zero and discard the usage of quite valuable repos, or you can copy them inside your project thus loosing the ability to upgrade it via composer, the only upgrade method left would be comparing files with github repo and merging changes manually, which would discard of any motivation of even using that php artisan app:name command.

So basically, I'm suggesting removing it at all or extending functionality, which would as well change App namespace in vendor repos and as well automatically would append a post install script to composer.json, which would rename App namespace after composer install.

### Steps To Reproduce:

Its not bug, its feature removal or upgrading offer.

**Comment by lagbox** [comment]
Changing stuff in vendor would be changing other packages code ... which is not something you do.

Tell the package developer of the package you are mad about to change what they are doing.

**Comment by juslintek** [comment]
Okay I flamed about this here: https://github.com/Laravel-Backpack/BackupManager/issues/27

**Comment by jamesgraham** [comment]
That package does not look fit for production.

**Comment by Dylan-DPC** [comment]
@juslintek this is a proposal, post it in laravel/internals repo

**Comment by themsaid** [comment]
Yes that isn't a bug, packages need to put in mind that the app namespace can be changed.

For suggestions and feature requests please use the https://github.com/laravel/internals repo.
